pyquil/tests/test_base_connection.py

```diff
@@ -0,0 +1,270 @@
+import pytest
+import os
+import requests_mock
+from configparser import ConfigParser
+from pyquil.api._base_connection import ForestSession
+from pyquil.api._config import PyquilConfig
+from pyquil.api._logger import UserMessageError
+import urllib.parse
+
+
+@pytest.fixture
+def successful_engagement_response():
+    return {
+        'engage': {
+            'success': True,
+            'message': "Good job, you engaged all by yourself",
+            'engagement': {
+                'type': 'RESERVATION',
+                'qpu': {
+                    'endpoint': 'tcp://fake.url:12345',
+                    'credentials': {
+                        'clientPublic': 'abc123',
+                        'clientSecret': 'abc123',
+                        'serverPublic': 'abc123'
+                    }
+                },
+                'compiler': {
+                    'endpoint': 'tcp://fake.url:12346'
+                },
+                'expiresAt': '9999999999.0'
+            }
+        }
+    }
+
+
+@pytest.fixture
+def failed_engagement_response():
+    return {
+        'engage': {
+            'success': False,
+            'message': "That did not work",
+            'engagement': None
+        }
+    }
+
+
+def fixture_path(path: str) -> str:
+    dir_path = os.path.dirname(os.path.realpath(__file__))
+    return os.path.join(dir_path, 'data', path)
```

Comment:
This function also appears in the config tests. Maybe move it into `tests/utils.py`